### PR TITLE
DDF-2728 Remove EHCache dependency from catalog ui

### PR DESCRIPTION
**Pull request by taz77**
#### What does this PR do?

Remove an unused dependency committed via https://github.com/codice/ddf-ui/pull/7

Description of the PR is nightly email notifications of queries in workspaces.

The manifests of builds show this library as not being used.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?

@clockard 
@brianfelix 
@mcalcote
@vinamartin 
@cruzj6

#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).

[UI](https://github.com/orgs/codice/teams/ui)

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).

@pklinef
@shaundmorris

#### How should this be tested? (List steps with links to updated documentation)

Attempt a full build. All tests should pass. The compiled DDF should be unziped and executed. Confirm that workspaces are working in the new search UI.

#### What are the relevant tickets?
[DDF-2728](https://codice.atlassian.net/browse/DDF-2728)

#### Screenshots (if appropriate)
N/A
#### Checklist:
- [x] Documentation Updated
- [x] Change Log Updated
- [x] Update / Add Unit Tests
- [x] Update / Add Integration Tests

### CODE REVIEW

**Comment by taz77** [comment]
I will squash the commits before merging.

File: catalog/ui/catalog-ui-search/pom.xml [hunk 1]
```diff
@@ -274,8 +264,6 @@
                             quartz-jobs,
                             c3p0,
                             terracotta-toolkit-runtime-ee,
```
Comment (shaundmorris) [inline catalog/ui/catalog-ui-search/pom.xml]: do we know if this `terracotta-toolkit-runtime-ee` is being used?

Comment (taz77) [inline catalog/ui/catalog-ui-search/pom.xml]: Sorry for the delayed response. I tried to remove that dependency as well. It didn't appear in the manifest to my knowledge but one of the itests fail. I have to sanity check this PR again tonight. I will test removing that as well, I can run parallel builds.

**Comment by pklinef** [comment]
@taz77 Remember to add the ticket DDF-2728 into the commit message when you squash the commits.

File: catalog/ui/catalog-ui-search/pom.xml (hunk 1 above)
Comment (pklinef) [inline catalog/ui/catalog-ui-search/pom.xml]: This should not be required and I would expect it to be a no-op as it is.

Comment (taz77) [inline catalog/ui/catalog-ui-search/pom.xml]: So far, removing this causes iTest failure. Trying to research why within a decent time frame.

**Comment by cruzj6** [comment]
I did I hero build on this, all itests and unit tests passed, build successful.

File: catalog/ui/catalog-ui-search/pom.xml (hunk 1 above)
Comment (taz77) [inline catalog/ui/catalog-ui-search/pom.xml]: SSO itests fail and CatalogSearchUI tests fail to setup as well. If it sounds acceptable to everyone here, push forward with this PR and write another ticket to look into this `terracotta-toolkit-runtime-ee` dependency.

**Comment by taz77** [comment]
Commits squashed and tagged.

**Pull request merged**